src/components/buttons/EdgeButton.tsx

```diff
@@ -74,233 +81,217 @@ export function EdgeButton(props: Props) {
-
-    retStyle.push(
-      customMarginPadding != null
-        ? customMarginPadding
-        : styles.touchContainerSpacing
-    )
+  // Sizing rules per variant:
+  const isTertiary = type === 'tertiary'
+  const visualHeightRem = isTertiary ? 1.5 : mini ? 2.0 : 3.0
+  const paddingXRem = isTertiary ? 0.75 : mini ? 1.25 : 1.5
+  const paddingYRem = isTertiary ? 0.0 : 0
 
-    return retStyle
-  }, [layout, customMarginPadding, styles])
+  const containerMargin = sidesToMargin(
+    mapSides(fixSides(marginRem, 0), theme.rem)
+  )
 
-  const visibleContainerStyle = React.useMemo(() => {
-    const retStyle: ViewStyle[] = [styles.visibleContainerCommon]
+  const opacity = disabled ? 0.3 : spinner || pending ? 0.7 : 1
 
-    if (layout === 'column') retStyle.push(styles.visibleContainerColumn)
-    if (layout === 'fullWidth') retStyle.push(styles.visibleContainerFullWidth)
-    if (layout === 'row') retStyle.push(styles.visibleContainerRow)
-    if (layout === 'solo') retStyle.push(styles.visibleContainerSolo)
-    if (type === 'tertiary') retStyle.push(styles.visibleContainerTertiary)
+  // Layout behavior for parent containers:
+  const layoutContainerStyle = React.useMemo<ViewStyle>(() => {
+    if (layout === 'row') {
+      return { width: '50%' }
+    }
+    if (layout === 'column' || layout === 'fullWidth') {
+      return {
+        alignSelf: 'stretch',
+        alignItems: 'stretch',
+        flexBasis: 'auto',
+        flexGrow: 0,
+        flexShrink: 0,
+        padding: layout === 'fullWidth' ? theme.rem(1) : undefined
+      }
+    }
+    // solo
+    return { alignItems: 'center', justifyContent: 'center' }
+  }, [layout, theme])
+
+  // Content row adjusted according to spinner prop value
+  const contentRowStyle = React.useMemo<ViewStyle>(
+    () => ({
+      ...styles.contentRow,
+      opacity: spinner ? 0 : 1
+    }),
+    [spinner, styles.contentRow]
+  )
 
-    retStyle.push(
-      mini
-        ? styles.visibleSizeMini
-        : type === 'tertiary'
-        ? styles.visibleSizeTertiary
-        : styles.visibleSizeDefault
-    )
-    retStyle.push({
-      opacity: disabled ? 0.3 : hideContent ? 0.7 : 1
-    })
+  // Tappable area overshoot
+  const hitSlop =
+    layout === 'solo'
+      ? {
+          top: theme.rem(0.5),
+          bottom: theme.rem(0.5),
+          // Expand horizontally so taps register across the full width
+          left: 10000,
+          right: 10000
+        }
+      : theme.rem(0.5)
+
+  const content = (
+    <View
+      style={[
+        styles.pillBase,
+        type !== 'tertiary' ? styles.pillSolid : styles.pillTertiary,
+        layout === 'column' || (layout === 'row' && !isTertiary)
+          ? styles.pillStretch
+          : null,
+        {
+          height: theme.rem(visualHeightRem),
+          paddingHorizontal: theme.rem(paddingXRem),
+          paddingVertical: theme.rem(paddingYRem),
+          opacity
+        }
+      ]}
+    >
```

Comment:
These styles can be moved directly to the EdgeTouchableOpacity. No need for the extra view here.

Reply:
Good catch, thanks